pkg/api/v1/types.go

```diff
@@ -1292,6 +1295,21 @@ type KeyToPath struct {
 	Mode *int32 `json:"mode,omitempty" protobuf:"varint,3,opt,name=mode"`
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string `json:"nodeName" protobuf:"bytes,1,opt,name=nodeName"`
+	// Fs represents a filesystem-based local volume
+	// +optional
```

Comment:
this says optional but is validates as a one-of.  @mbohlool do we have a preferred way to decorate "one-of" groups?